src/pages/academy/game/Game.tsx

```diff
@@ -50,9 +52,93 @@ function Game() {
     }
   }, [session, achievements, goals]);
 
+  // This is a custom hook imported from @mantine/hooks that handles the fullscreen logic
+  // It returns a ref to attach to the element that should be fullscreened,
+  // a function to toggle fullscreen and a boolean indicating whether the element is fullscreen
+  const {
+    ref: fullscreenRef,
+    toggle: toggleFullscreen,
+    fullscreen: isFullscreen
+  } = useFullscreen<HTMLDivElement>();
+
+  // This function is a wrapper around toggleFullscreen that also locks the screen orientation
+  // to landscape when entering fullscreen and unlocks it when exiting fullscreen
+  const enhancedToggleFullscreen = async () => {
+    toggleFullscreen();
+
+    if (window.screen.orientation) {
+      if (!isFullscreen) {
+        window.screen.orientation.lock('landscape');
+      } else {
+        window.screen.orientation.unlock();
+      }
+    }
+  };
+
+  const gameDisplayRef = React.useRef<HTMLDivElement | null>(null);
+
+  // This function sets the gameDisplayRef and also calls the ref callback from useFullscreen
+  // to attach the fullscreen logic to the game display element
+  const setGameDisplayRefs = React.useCallback(
+    (node: HTMLDivElement | null) => {
+      // Refs returned by useRef()
+      gameDisplayRef.current = node;
+
+      // Ref callback from useFullscreen
+      fullscreenRef(node);
+    },
+    [fullscreenRef]
+  );
+
+  // Logic for the fullscreen button to dynamically adjust its size, position and padding
+  // based on the size of the game display.
+  const [iconSize, setIconSize] = React.useState(0);
+  const [iconLeft, setIconLeft] = React.useState('0px');
+  const [iconPadding, setIconPadding] = React.useState('0px');
+
+  React.useEffect(() => {
+    const handleResize = () => {
+      if (gameDisplayRef.current) {
+        const aspectRatio = 16 / 9;
+        const height = gameDisplayRef.current.offsetHeight;
+        const width = gameDisplayRef.current.offsetWidth;
+        const size = height / 40;
+        const padding = height / 50;
+        const leftOffset =
+          isFullscreen || height * aspectRatio > width ? 0 : (width - height * aspectRatio) / 2;
+        setIconSize(size);
+        setIconPadding(`${padding}px`);
+        setIconLeft(`${leftOffset}px`);
```

Comment:
I don't get it:

1. Why are `iconSize`, `iconPadding`, `iconLeft` independent state variables? Aren't they all derived state?
2. What does the game height and width have anything to do with the icon positioning?
3. What is the aspect ratio for?